<?xml version="1.0"?>
<launch>
    <!-- declare arg to be passed in -->
    <arg name="team_side" default="home" />
    <arg name="sim_mode" default="true" />
    <arg name="two_robots" default="true" />

    <!-- Outputs to either real world or simulation -->
    <arg name="vel_cmd_ally1" default="vel_cmds" />
    <arg name="vel_cmd_ally2" default="vel_cmds" />
    <arg name="kick_ally1" default="kick" />
    <arg name="kick_ally2" default="kick" />

    <group ns="teamrocket_$(arg team_side)">

        <!-- Set ROS parameters that the code can access -->
        <param name="team_side" type="string" value="$(arg team_side)" />

        <!-- start up ally1 -->
        <group ns="ally1">
            <!-- Start the robot up -->
            <include file="$(find teamrocket)/launch/robot.launch">
                <arg name="my_number" value="1" />
                <arg name="ally_number" value="2" />
                <arg name="team_side" value="$(arg team_side)" />
                <arg name="orienter_path"
                     value="/teamrocket_$(arg team_side)/orienter" />
            </include>
        </group>

        <!-- start up ally2 -->
        <group ns="ally2" if="$(arg two_robots)">
            <!-- Start the robot up -->
            <include file="$(find teamrocket)/launch/robot.launch">
                <arg name="my_number" value="2" />
                <arg name="ally_number" value="1" />
                <arg name="team_side" value="$(arg team_side)" />
                <arg name="orienter_path"
                     value="/teamrocket_$(arg team_side)/orienter" />
            </include>
        </group>

        <!-- Orienter -->
        <node name="orienter" pkg="teamrocket" type="orienter_node.py" output="screen">
        </node>

    </group>
</launch>
